providers/src/airflow/providers/google/ads/hooks/ads.py

```diff
@@ -267,24 +270,36 @@ def _update_config_with_secret(self, secrets_temp: IO[str]) -> None:
         self.google_ads_config["json_key_file_path"] = secrets_temp.name
 
     def _search(
-        self, client_ids: list[str], query: str, page_size: int = 10000, **kwargs
+        self, client_ids: list[str], query: str, page_size: int | None = None, **kwargs
     ) -> list[GoogleAdsRow]:
         """
         Pull data from the Google Ads API.
 
         :param client_ids: Google Ads client ID(s) to query the API for.
         :param query: Google Ads Query Language query.
-        :param page_size: Number of results to return per page. Max 10000.
+        :param page_size: Number of results to return per page. Max 10000 (for version 16 and 16.1)
+            This parameter deprecated. After March 01, 2025, it will be removed.
 
         :return: Google Ads API response, converted to Google Ads Row objects
         """
         service = self._get_service
 
+        extra_req_params = {}
+        if self.api_version == "v16":  # TODO: remove this after deprecation removal for page_size parameter
+            extra_req_params["page_size"] = page_size or 10000
+        else:
+            if page_size:
+                warnings.warn(
+                    "page_size parameter for the GoogleAdsHook.search and "
+                    "GoogleAdsHook.search_proto_plus method is deprecated and will be removed "
+                    "after March 01, 2025.",
+                    AirflowProviderDeprecationWarning,
+                    stacklevel=2,
+                )
+
```

Comment:
Will the v16 reach the end of life by March 01, 2025, so we could remove this parameter safely?

Reply:
According to this page: https://developers.google.com/google-ads/api/docs/sunset-dates#feature_deprecations, it says after February in the upper section of that page mentions about the 5th of February but I am not entirely sure it will happen exactly on 5th of February. Maybe I can change this to February 05, 2025 and we can remove this on February.